noxfile.py

```diff
@@ -0,0 +1,261 @@
+        silent=True,
+    )
+
+
+@nox.session(reuse_venv=True)
+def isort(session: Session) -> None:
+    """Run isort on cachi2 and tests directories and noxfile.py"""
+    _install_requirements(session)
+    session.run(
+        "isort",
+        "--check",
+        "--diff",
+        "--color",
+        "cachi2",
+        "tests",
+        "noxfile.py",
+        *session.posargs,
+        silent=True,
+    )
+
+
+@nox.session(reuse_venv=True)
+def mypy(session: Session) -> None:
+    """Run mypy on cachi2 and tests directories"""
+    _install_requirements(session)
+    session.run(
+        "mypy",
+        "--install-types",
+        "--non-interactive",
+        "cachi2",
+        "tests",
+        *session.posargs,
+        silent=True,
+    )
+
+
+@nox.session(name="unit-tests", reuse_venv=True, python=["3.9", "3.10", "3.11", "3.12"])
+def unit_tests(session: Session) -> None:
+    """Run unit tests and generate coverage report"""
+    _install_requirements(session)
+    session.run(
+        "pytest",
+        "--log-level=DEBUG",
+        "--cov=cachi2",
+        "--cov-config=pyproject.toml",
+        "--cov-report=term",
+        "--cov-report=html",
+        "tests/unit",
+        *session.posargs,
+    )
+
+
+@nox.session(name="coverage-html", reuse_venv=True)
+def coverage_html(session: Session) -> None:
+    """Open the coverage report in the browser"""
+    session.install("coverage")
+    session.run("coverage", "html")
+    session.run("xdg-open", "htmlcov/index.html", external=True)
+
+
+def _run_integration_tests(session: Session, env: dict[str, str]) -> None:
+    _install_requirements(session)
+    default_env = {
+        "CACHI2_TEST_NETRC_CONTENT": "machine 127.0.0.1 login cachi2-user password cachi2-pass",
+        "CACHI2_TEST_LOCAL_PYPISERVER": "true",
+        "CACHI2_TEST_LOCAL_DNF_SERVER": "true",
+    }
+    default_env.update(env)
+    session.run(
+        "pytest",
+        "--log-cli-level=WARNING",
+        "tests/integration",
+        *session.posargs,
+        env=default_env,
+    )
+
+
+@nox.session(name="integration-tests", reuse_venv=True)
+def integration_tests(session: Session) -> None:
+    """Run integration tests only for the affected code base in the current branch"""
+    _run_integration_tests(
+        session,
+        {},
+    )
+
+
+@nox.session(name="all-integration-tests", reuse_venv=True)
+def all_integration_tests(session: Session) -> None:
+    """Run all integration tests that are available"""
+    _run_integration_tests(
+        session,
+        {
+            "CACHI2_RUN_ALL_INTEGRATION_TESTS": "true",
+        },
+    )
+
+
+@nox.session(name="generate-test-data", reuse_venv=True)
+def generate_test_data(session: Session) -> None:
+    """Run all integration tests that are available and update SBOMs"""
+    _run_integration_tests(
+        session,
+        {
+            "CACHI2_RUN_ALL_INTEGRATION_TESTS": "true",
+            "CACHI2_GENERATE_TEST_DATA": "true",
+        },
```

Comment:
Similarly here, noxfile is IMO opaque, meaning that nobody will bother looking into it unless they need to add or fix a target, so I think documenting the usage of these env vars to adjust the behaviour of the test target is IMO more transparent than essentially defining "sugar coated" targets which just hinder transparency; I think less is more in this case.